src/main/scala/com/databricks/spark/avro/avroRelation2/AvroRelation2.scala

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2014 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.avro.avroRelation2
+
+import java.io.FileNotFoundException
+import java.util.zip.Deflater
+
+import com.databricks.spark.avro.{AvroSaver, SchemaConverters}
+import com.google.common.base.Objects
+import org.apache.avro.Schema.Field
+import org.apache.avro.file.{DataFileConstants, DataFileReader, FileReader}
+import org.apache.avro.generic.{GenericDatumReader, GenericRecord}
+import org.apache.avro.mapred.{AvroOutputFormat, FsInput}
+import org.apache.avro.mapreduce.AvroJob
+import org.apache.avro.{Schema, SchemaBuilder}
+import org.apache.hadoop.fs.{FileStatus, FileSystem, Path}
+import org.apache.hadoop.mapreduce.Job
+import org.apache.spark.Logging
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.sources._
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{Row, SQLContext}
+import parquet.hadoop.util.ContextUtil
+
+import scala.collection.Iterator
+import scala.collection.JavaConversions._
+
+class AvroRelation2(override val paths: Array[String],
+                   private val maybeDataSchema: Option[StructType],
+                   override val userDefinedPartitionColumns: Option[StructType],
+                   private val parameters: Map[String, String])
+                  (@transient val sqlContext: SQLContext) extends HadoopFsRelation with Logging {
+
+  private val IgnoreFilesWithoutExtensionProperty = "avro.mapred.ignore.inputs.without.extension"
+  private val recordName = parameters.getOrElse("recordName",
+    AvroSaver.defaultParameters.get("recordName").get)
```

Comment:
Instead of `.get(...).get` just use the `apply` method or consider not keeping the defaults in a map (so you get compile time checks).